complex/complex.go

```diff
@@ -2,7 +2,7 @@
 // Use of this code is governed by a BSD-style
 // license that can be found in the LICENSE file
 
-package cfloats
+package complex
```

Comment:
seems a bit akward to hide a builtin function with a package name.

perhaps name it `cmplxs` ? (to not collide with `math/cmplx`)
or `c128s` ?

Reply:
OK, I can change it to cmplxs. That's not a problem. I'm just trying to keep it as something more similar to floats, since it is basically a complex number version of that.

Reply:
I am not good with package names (so please wait for others to chime in and add a bit of paint on that shed).
I was just raising this potential identifier collision.

Reply:
After thinking about this, I think the right approach is to just extend the existing floats package with Cx functions. This is probably the most straight forward, and ties in with the implementation of most of the rest of gonum. For some reason, I was thinking about how BLAS has f64 and c128 separate. But that was probably wasn't the right thinking. If anyone thinks otherwise, I'm going to move forward with moving these functions into the existing floats package.

Reply:
Please leave them separate.

Reply:
OK, will do. My main concern was with the consistency of how the other sections, mat & num, have float and complex together.

Do you still want to keep it completely separate, or make it a subdirectory of floats. Meaning floats/cmplxs

Reply:
The mat package is a special case that arises because of no-cyclic imports. The num/... packages don't do this; each of those packages deals with one type of number.

I would really like for @btracey to chime in for this as he was the architect of floats and so has the best overview for this.

Reply:
`// TODO(btracey): Move these into a cmplxs if/when we have one.`

Found this line in cmatrix.go while working on adding functionality to the CMatrix & CDense types. Looks like cmplxs is what he had planned on naming it. I'll go with that as a completely separate directory.